packages/cadl-ranch-specs/http/azure/arm/models/resources/toplevel.tsp

```diff
@@ -0,0 +1,214 @@
+  Expected query parameter: api-version=2023-12-01-preview
+
+  Expected response body:
+  ```json
+  {
+    "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.Arm.Models.Resources/topLevelTrackedResources/top",
+    "name": "top",
+    "type": "topLevel",
+    "location": "eastus",
+    "properties":{
+      "description": "valid",
+      "provisioningState": "Succeeded"
+    },
+    "systemData": {
+      "createdBy": "AzureSDK",
+      "createdByType": "User",
+      "createdAt": <any date>,
+      "lastModifiedBy": "AzureSDK",
+      "lastModifiedAt": <any date>,
+      "lastModifiedByType": "User",
+    }
+  }
+  ```
+  """)
+  get is ArmResourceRead<TopLevelTrackedResource>;
+
+  @scenario
+  @scenarioDoc("""
+  Resource PUT operation.
+  Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.Arm.Models.Resources/topLevelTrackedResources/top
+  Expected query parameter: api-version=2023-12-01-preview
+  Expected request body:
+  ```json
+  {
+    "location": "eastus",
+    "properties": {
+      "description": "valid"
+    }
+  }
+  ```
+  Expected response body:
+  ```json
+  {
+    "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.Arm.Models.Resources/topLevelTrackedResources/top",
+    "name": "top",
+    "type": "topLevel",
+    "location": "eastus",
+    "properties": {
+      "description": "valid",
+      "provisioningState": "Succeeded"
+    },
+    "systemData": {
+      "createdBy": "AzureSDK",
+      "createdByType": "User",
+      "createdAt": <any date>,
+      "lastModifiedBy": "AzureSDK",
+      "lastModifiedAt": <any date>,
+      "lastModifiedByType": "User",
+    }
+  }
+  ```
+  """)
+  createOrReplace is ArmResourceCreateOrReplaceAsync<TopLevelTrackedResource>;
+
+  @scenario
+  @scenarioDoc("""
+  Resource PATCH operation.
+  Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.Arm.Models.Resources/topLevelTrackedResources/top
+  Expected query parameter: api-version=2023-12-01-preview
+  Expected request body:
+  ```json
+  {
+    "properties": {
+      "description": "valid2"
+    }
+  }
+  ```
+  Expected response body:
+  ```json
+  {
+    "id": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.Arm.Models.Resources/topLevelTrackedResources/top",
+    "name": "top",
+    "type": "topLevel",
+    "location": "eastus",
+    "properties":{
+      "description": "valid2",
+      "provisioningState": "Succeeded"
+    },
+    "systemData": {
+      "createdBy": "AzureSDK",
+      "createdByType": "User",
+      "createdAt": <any date>,
+      "lastModifiedBy": "AzureSDK",
+      "lastModifiedAt": <any date>,
+      "lastModifiedByType": "User",
+    }
+  }
+  ```
+  """)
+  update is ArmResourcePatchAsync<TopLevelTrackedResource, TopLevelTrackedResourceProperties>;
```

Comment:
this will cause splitting of first layer model. i'm not sure if we need to change to use `custompatch` or wait for tsp fix.

Reply:
If this is used by a real service (I remember there is `ArmResourcePatchSync`) and it is the recommended over `custompatch`, we should have this test.

Split or not is a question on either typespec or emitter, but before this is decided/addressed, we would still need to release for the real service.